src/NumoBundle/Controller/EventController.php

```diff
@@ -269,21 +301,108 @@ public function showAction(Request $request, $id)
+     * @Method({"GET", "POST"})
+     */
+    public function editAwitAction(Request $request, Event $event)
+    {
+        // --- Note : les images sont gerees par des eventlisteners
+        $imgDir = $this->getParameter('img_event_dir');
+        $em = $this->getDoctrine()->getManager();
+        $oldImage = $event->getImage();
+        $originalEvtDates = new ArrayCollection();
+        foreach ($event->getEvtDates() as $evtDate) {
+            $originalEvtDates->add($evtDate);
+        }
+        $form = $this->createForm(EventType::class, $event);
+        $form->handleRequest($request);
 
+        if ($form->isSubmitted() && $form->isValid()) {
+            // --- adaptation de la liste des dates
+            foreach ($originalEvtDates as $evtDate){
+                if (false === $event->getEvtDates()->contains($evtDate)) {
+                    $em->remove($evtDate);
+                }
+            }
+            $event
+                ->setAuthor($this->getUser())
+                ->setCreationDate(new \datetime());
+            $em->flush();
+
+            return $this->redirectToRoute('event_show_await', ['id' => $event->getId()]);
+        }
+
+        return $this->render('NumoBundle:event:editAwait.html.twig', [
+            'event' => $event,
+            'imgDir' => $imgDir,
+            'eventId' => $event->getId(),
+            'oldImage' => $oldImage,
+            'form' => $form->createView(),
+        ]);
+
+    }
 
     /**
-     * Displays a form to edit an existing event entity.
+     * Deletes an awaiting event.
      *
-     * @Route("/{id}/edit", name="event_edit")
-     * @Method({"GET", "POST"})
+     * @Route("/deleteawait/{id}", name="event_delete_await")
+     * @Method({"GET","POST"})
      */
-    public function editAction(Request $request, Event $event)
+    public function deleteAwaitAction(Request $request, Event $event)
     {
+        // --- Note : les images sont gerees par des eventlisteners
+        $imgDir = $this->getParameter('img_event_dir');
+        $form = $this
+            ->createFormBuilder()
+            ->add('delete', SubmitType::class, ['label' => 'Supprimer'])
+            ->getForm();
+        $form->handleRequest($request);
+        // --- generation des tableaux dates pour affichage
+        $oldDates = $newDates = [];
+        $dateRef = new \DateTime();
+        foreach ($event->getEvtDates() as $evtD) {
+            $evtDate = [
+                'evtDate' => $evtD->getEvtDate()->format('Y-m-d'),
+                'timeStart' => $evtD->getTimeStart()->format('H:i'),
+                'timeEnd' => $evtD->getTimeEnd()->format('H:i')
+            ];
+            if ($evtDate['evtDate'] < $dateRef->format('Y-m-d')) {
+                $oldDates[] = $evtDate;
+            } else {
+                $newDates[] = $evtDate;
+            }
+        }
+        // --- definition de la route de retour
+        if ($this->getUser()->getRoles()[0] == 'ROLE_MODERATOR') {
```

Comment:
non, car le user pourrait avoir le role moderateur mais pas en position 0
soit tu fait in_array('ROLE_MODERATOR', $this->getUser->getRoles())
ou encore mieux (à retester pour la syntaxe)
if ($this->isGranted('ROLE_MODERATOR', $this->getUser())